go-controller/pkg/factory/factory.go

```diff
@@ -116,6 +151,53 @@ func (i *informer) removeHandler(handler *Handler) error {
 	return nil
 }
 
+func (i *informer) processEvents(events chan *event, stopChan <-chan struct{}) {
+	for {
+		select {
+		case e, ok := <-events:
+			if !ok {
```

Comment:
@danwinship tried a couple things there, but eventually decided to have a new lock for the event array that is only used when writing to the array and shutting it down, so we can check for a nil entry in the array and thus know the channel is closed. Otherwise there's a race between wf.shutdown() and the shared informer goroutines that can cause a panic due to write to a closed channel.